interp/type.go

```diff
@@ -841,11 +841,19 @@ func (t *itype) methods() methodSet {
 				res[m.ident] = m.typ.TypeOf().String()
 			}
 		}
-	default:
-		for _, m := range t.method {
-			res[m.ident] = m.typ.TypeOf().String()
+		for k, v := range t.val.methods() {
```

Comment:
Reminder as per chat: this loop seems to have the counter-effect of the filter above. Also, we need to check whether L839 and L844 have different effects, to pick the right one that we want to keep.

Reply:
I removed this code block which is useless. Thanks for noticing.